instrumentation/tomcat/tomcat-jdbc/src/main/java/io/opentelemetry/javaagent/instrumentation/tomcat/jdbc/TomcatConnectionPoolMetrics.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.tomcat.jdbc;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.metrics.ObservableLongUpDownCounter;
+import io.opentelemetry.instrumentation.api.metrics.db.DbConnectionPoolMetrics;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import org.apache.tomcat.jdbc.pool.DataSourceProxy;
+
+public final class TomcatConnectionPoolMetrics {
+
+  private static final OpenTelemetry openTelemetry = GlobalOpenTelemetry.get();
+  private static final String INSTRUMENTATION_NAME = "io.opentelemetry.tomcat-jdbc";
+
+  private static final Map<DataSourceProxy, List<ObservableLongUpDownCounter>> dataSourceMetrics =
```

Comment:
does @laurit's comment from other PR make sense here?

```
  // a weak map does not make sense here because each Meter holds a reference to the connection pool
  // none of the UniversalConnectionPool implementations contain equals()/hashCode(), so it's safe
  // to keep them in a plain ConcurrentHashMap
```

Reply:
Sure,  I'll put it back. The reason I omitted it is that I typically don't doc all the things that a piece of code is not. :)  I grok that WeakMap is favored in instrumentation, though, so it probably does make sense to include it for our future selves.